odoo/models.py

```diff
@@ -2502,6 +2515,8 @@ def _read_group_format_result(self, rows_dict, lazy_groupby):
                                 (field_name, '>=', range_start),
                                 (field_name, '<', range_end),
                         ]
+                    elif value and granularity in RELATIVE_GROUP_TIME_GRANULARITY:
```

Comment:
In case you group by hour, you can get value `0`, which is falsy.
```suggestion
                    elif value is not None and granularity in RELATIVE_GROUP_TIME_GRANULARITY:
```